src/algebra/group_power.lean

```diff
@@ -532,6 +532,25 @@ lemma smul_le_smul_of_le_right {a b : α} (hab : a ≤ b) : ∀ i : ℕ, i • a
 
 end add_monoid
 
+namespace canonically_ordered_semiring
```

Comment:
There's a thing called an `ordered_semiring`. Could the existing theorems be generalised to an `ordered_semiring` rather than this duplication?

Reply:
`ennreal` is not an `ordered_semiring` because `1 < ∞`, `0 < ∞` but `1 * ∞ = ∞ * ∞`.